src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -392,16 +384,39 @@ protected void parseCreateField(ParseContext context) throws IOException {
 
     protected void parseCreateField(ParseContext context, int dimension) throws IOException {
 
-        if (!KNNSettings.isKNNPluginEnabled()) {
-            throw new IllegalStateException("KNN plugin is disabled. To enable " + "update knn.plugin.enabled setting to true");
+        validateIfKNNPluginEnabled();
+        validateIfCircuitBreakerTriggered();
+
+        Optional<float[]> arrayOptional = getFloatsFromContext(context, dimension);
+
+        if (!arrayOptional.isPresent()) {
+            return;
         }
+        final float[] array = arrayOptional.get();
+        VectorField point = new VectorField(name(), array, fieldType);
 
+        context.doc().add(point);
+        if (fieldType.stored()) {
+            context.doc().add(new StoredField(name(), point.toString()));
+        }
+        context.path().remove();
+    }
+
+    void validateIfCircuitBreakerTriggered() {
         if (KNNSettings.isCircuitBreakerTriggered()) {
             throw new IllegalStateException(
                 "Indexing knn vector fields is rejected as circuit breaker triggered." + " Check _opendistro/_knn/stats for detailed state"
             );
         }
+    }
 
+    void validateIfKNNPluginEnabled() {
+        if (!KNNSettings.isKNNPluginEnabled()) {
+            throw new IllegalStateException("KNN plugin is disabled. To enable " + "update knn.plugin.enabled setting to true");
+        }
+    }
+
+    Optional<float[]> getFloatsFromContext(ParseContext context, int dimension) throws IOException {
```

Comment:
This part will be common with Lucene field mapper, extract it to a method to avoid code duplication